Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,145 @@ protected override void ProcessRecord()
             }
         }
     }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Creates a new empty table in the specified project and dataset.
+    /// </para>
+    /// <para type="description">
+    /// Creates a new empty table in the specified dataset. A Table can be supplied by object via 
+    /// the pipeline or the -InputObject parameter, or it can be instantiated by value with the 
+    /// flags below. A Dataset must be specified, either by passing string ID or a Dataset object 
+    /// with the -Dataset parameter, or passing a Dataset object in through the pipeline. If a 
+    /// Dataset is sent in on the pipeline, you must use the “ByValue” parameter set. If no Project 
+    /// is specified, the default project will be used. This cmdlet returns a Table object.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> New-GcbqTable</code>
+    ///   <para>This does a thing</para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/bigquery/docs/reference/rest/v2/tables)">
+    /// [BigQuery Tables]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "GcbqTable")]
+    public class NewGcbqTable : GcbqCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByObject = "ByObject";
+            public const string ByValues = "ByValue";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be inserted.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        public Table InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to put the table in. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        override public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Dataset that you would like to add to. This field takes strings, Datasets, and DatasetRefrences.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByValues)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset), Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
```

Comment:
You probably need a third ParameterSet to take a Dataset object rather than a string, because the `DatasetReference` will specify the project for you.